### PR TITLE
Extracted out multiple methods of add method

### PR DESCRIPTION
**Pull request by cartothemax**
Work for issue #442

Detangled some of the logic that was tied up in the if statement by
extracting the logic determining the pusher_type. Created
`setup_allyship`, `remove_allyship_request`, and
`send_allyship_notifications` to help clarify what all is a part of the
add method within the allies_controller.

### CODE REVIEW

**Comment by cartothemax** [comment]
Looks like I have some CodeClimate issues to take a look at.  Will take a look tonight!

File: app/controllers/allies_controller.rb [hunk 1]
```diff
@@ -19,65 +19,77 @@ def add
...
     data = JSON.generate(
       user: current_user.name,
       userid: current_user.id,
       uid: current_user.uid,
       type: pusher_type,
-      uniqueid: uniqueid
+      uniqueid: unique_id
     )
 
+    send_allyship_notifications(unique_id, data, pusher_type, ally_id)
```
Comment (julianguyen) [inline app/controllers/allies_controller.rb]: Looks like you have to update this method to take four arguments! Right now it's only taking 3.

```
ArgumentError:
  wrong number of arguments (given 4, expected 3)
./app/controllers/allies_controller.rb:83:in `send_allyship_notifications'
./app/controllers/allies_controller.rb:80:in `handle_allyship_notifications'
./app/controllers/allies_controller.rb:28:in `add'
./app/controllers/application_controller.rb:31:in `block in with_timezone'
./app/controllers/application_controller.rb:31:in `with_timezone'
./spec/controllers/allies_controller_spec.rb:59:in `block (4 levels) in <top (required)>'
./spec/controllers/allies_controller_spec.rb:80:in `block (5 levels) in <top (required)>'
```

Comment (cartothemax) [reply app/controllers/allies_controller.rb]: Thanks for the review! Got this cleaned up and made the tests green again 😄

**Comment by cartothemax** [comment]
Hey 👋 @julianguyen this PR is all ready again when you get a chance.  Thanks!

**Pull request merged**